src/source_map_support.js

```diff
@@ -103,7 +103,7 @@ if (!isDataURI(wasmSourceMapFile)) {
 
 function getSourceMap() {
   var buf = readBinary(wasmSourceMapFile);
-  return JSON.parse(UTF8ArrayToString(buf, 0, buf.length));
```

Comment:
Is this change NFC? Before we would read `buf.length` bytes at most, but after we read without limit until the end of a UTF8 string. But is wasmSourceMapFile guaranteed to be encoded that way (including a null terminator)?

Reply:
The third argument here is `maxBytesToRead` which means will read at most `buf.length`. However `UTF8ArrayToString` will always stop when it reads a null byte I believe

Reply:
But is there a null byte in this input? That is what concerns me. It is reading a source map file, which I don't think is encoded that way.

Reply:
Oh I assumed UTF8ArrayToString would not read read past the end of the input array!  I don't see that check though.  That's strange.

Reply:
Oh wait I think its fine because if you read of the end of a array or buffer in JS you get `undefined` back, so every buffer is effectively null terminated.

Reply:
Ah, good point. That looks safe then.